src/main/java/org/jabref/gui/worker/CitationStyleToClipboardWorker.java

```diff
@@ -39,29 +39,33 @@
     private final String style;
     private final String previewStyle;
     private final CitationStyleOutputFormat outputFormat;
+    private final DialogService dialogService;
+    private final ClipBoardManager clipBoardManager;
 
-
-    public CitationStyleToClipboardWorker(BasePanel basePanel, CitationStyleOutputFormat outputFormat) {
+    public CitationStyleToClipboardWorker(BasePanel basePanel, CitationStyleOutputFormat outputFormat, DialogService dialogService, ClipBoardManager clipBoardManager, PreviewPreferences previewPreferences) {
         this.basePanel = basePanel;
         this.selectedEntries = basePanel.getSelectedEntries();
-        PreviewPreferences previewPreferences = Globals.prefs.getPreviewPreferences();
         this.style = previewPreferences.getPreviewCycle().get(previewPreferences.getPreviewCyclePosition());
-        this.previewStyle = Globals.prefs.getPreviewPreferences().getPreviewStyle();
+        this.previewStyle = previewPreferences.getPreviewStyle();
         this.outputFormat = outputFormat;
+        this.clipBoardManager = clipBoardManager;
 
-        basePanel.frame().setStatus(Localization.lang("Copying..."));
+        this.dialogService = dialogService;
+        dialogService.notify(Localization.lang("Copying..."));
+
+        BackgroundTask.wrap(this::loadCitationStyles).onFailure(ex -> LOGGER.error("Error while copying citations to the clipboard", ex))
```

Comment:
I guess you need to also run the background task (i.e call `execute()`). But I don't like the fact that the constructor has the side-effect of running a new thread. Thus, I would propose to let `CitationStyleToClipboardWorker` inherit from `BackgroundTask` and rename `loadCitationStyles` to `call` (so that it overrides the corresponding method in background task). Then the statement above in basepanel would become `new CitationStyleToClipboardWorker(...).executeWith(Globals.taskExecutor)`.